src/components/draft/editor/MonacoEditor.tsx

```diff
@@ -0,0 +1,107 @@
+import Editor from '@monaco-editor/react';
+import { useTheme } from '@mui/material';
+import useEditorStore, {
+    editorStoreSelectors,
+} from 'components/draft/editor/Store';
+import { debounce } from 'lodash';
+import * as monacoEditor from 'monaco-editor/esm/vs/editor/editor.api';
+import { useRef } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { supabaseClient, TABLES } from 'services/supabase';
+import { useQuery, useSelectSingle } from 'supabase-swr';
+
+type Props = {
+    currentCatalog: string;
+    draftId: string;
+};
+
+interface DraftSpec {
+    catalog_name: string;
+    spec_type: string;
+    spec_patch: string;
+    draft_id: string;
+}
+
+const DRAFT_SPECS_QUERY = `
+    catalog_name,
+    spec_type,
+    spec_patch,
+    draft_id
+`;
+
+function MonacoEditor({ draftId }: Props) {
+    const theme = useTheme();
+
+    const currentCatalog = useEditorStore(editorStoreSelectors.currentCatalog);
+
+    const draftSpecQuery = useQuery<DraftSpec>(
+        TABLES.DRAFT_SPECS,
+        {
+            columns: DRAFT_SPECS_QUERY,
+            filter: (query) =>
+                query
+                    .eq('draft_id', draftId)
+                    .eq('catalog_name', currentCatalog ? currentCatalog : ''),
+        },
+        [currentCatalog]
+    );
+    const { data: draftSpec, mutate } = useSelectSingle(draftSpecQuery, {
+        revalidateOnMount: true,
+    });
+
+    const editorRef = useRef<monacoEditor.editor.IStandaloneCodeEditor | null>(
+        null
+    );
+
+    const handlers = {
+        change: debounce(async () => {
```

Comment:
Wait for the user to stop typing for a second before saving changes to the server.